### PR TITLE
examples: enable debug logs

### PR DESCRIPTION
**Pull request by maryamtahhan**
Configure log level for examples and publish errors from the xsk APIs.

### CODE REVIEW

File: examples/l3fwd-graph/parse-args.c [hunk 1]
```diff
@@ -191,10 +193,11 @@ parse_args(int argc, char **argv)
     struct option lgopts[] = {{NULL, 0, 0, 0}};
     int opt, option_index, flags = 0;
     char json_file[1024] = {0};
+    char log_level[16]   = {0};
 
     /* Parse the input arguments. */
     for (;;) {
-        opt = getopt_long(argc, argv, "hc:dCDPV", lgopts, &option_index);
+        opt = getopt_long(argc, argv, "hc:dCDPVL", lgopts, &option_index);
```
Comment (KeithWiles) [inline examples/l3fwd-graph/parse-args.c:200]: Missing a ':' after the 'L'

File: examples/cnet-quic/parse-args.c [hunk 2]
```diff
@@ -231,7 +233,7 @@ parse_args(int argc, char **argv)
 
     /* Parse the input arguments. */
     for (;;) {
-        opt = getopt_long(argc, argvopt, "hc:s:dCDPVUu", lgopts, &option_index);
+        opt = getopt_long(argc, argvopt, "hc:s:dCDPVUuL", lgopts, &option_index);
```
Comment (KeithWiles) [inline examples/cnet-quic/parse-args.c:236]: Missing a ':' after the 'L'

File: lib/include/cne_strings.h [hunk 3]
```diff
@@ -260,6 +260,20 @@ cne_strqtok(char *str, const char *delim, char *argv[], int maxtokens)
...
+/**
+ * Convert characters in \p str to lowercase.
+ *
+ * @param str
+ *   String to convert to lowercase
+ */
+static __inline__ void
+cne_strtolower(char *str)
+{
+    for (int i = 0; i <= (int)(strlen(str)); i++) {
```
Comment (KeithWiles) [inline lib/include/cne_strings.h:272]: Single line for or if statements do not need {}

**Review by KeithWiles — CHANGES_REQUESTED** [review]
LGTM with a few minor nits and issues.

Thank you.

**Comment by maryamtahhan** [comment]
no worries - I will fix them up

File: examples/cndpfwd/parse-args.c [hunk 4]
```diff
@@ -252,18 +253,19 @@ static void
 print_usage(char *prog_name)
 {
     cne_printf("Usage: %s [-h] [-c json_file] [-b burst] <mode>\n"
-               "  <mode>         Mode types [drop | rx-only], tx-only, [lb | loopback], fwd, "
+               "  <mode>          Mode types [drop | rx-only], tx-only, [lb | loopback], fwd, "
```
[the remark this answers is not in the document]
Comment (maryamtahhan) [reply examples/cndpfwd/parse-args.c:256]: it's aligned with the text below

File: lib/include/cne_strings.h (hunk 3 above)
Comment (maryamtahhan) [reply lib/include/cne_strings.h:272]: oops - I missed this I will fix that

File: lib/include/cne_strings.h [hunk 5]
```diff
@@ -260,6 +260,39 @@ cne_strqtok(char *str, const char *delim, char *argv[], int maxtokens)
...
+ *   String to convert to uppercase
+ *
+ * @return
+ *   Upper case string
+ */
+static __inline__ char *
+cne_strtoupper(char *str)
+{
+    for (int i = 0; i <= (int)(strlen(str)); i++) {
+        str[i] = toupper(str[i]);
```
Comment (maryamtahhan) [inline lib/include/cne_strings.h:290]: same here

File: examples/l3fwd-graph/parse-args.c [hunk 6]
```diff
@@ -174,14 +175,15 @@ static void
...
-               "  -P             JCFG debug parsing\n"
-               "  -h             Display the help information\n",
+               "  <mode>          Mode types drop, tx-only, or [lb | loopback]\n"
+               "  -c <json-file>  The JSON configuration file\n"
+               "  -C              Wait on unix domain socket for JSON or JSON-C file\n"
+               "  -d              More debug stats are displayed\n"
+               "  -D              JCFG debug decoding\n"
+               "  -V              JCFG information verbose\n"
+               "  -P              JCFG debug parsing\n"
+               "  -L [info|debug] Enable debug or info Level Logging\n"
```
Comment (maryamtahhan) [inline examples/l3fwd-graph/parse-args.c:185]:
```suggestion
               "  -L [level] Enable a logging level\n"
```

Comment (KeithWiles) [reply examples/l3fwd-graph/parse-args.c:185]: Seems reasonable to change the help text.

File: lib/core/log/cne_log.h [hunk 7]
```diff
@@ -47,6 +64,17 @@ enum {
  */
 CNDP_API void cne_log_set_level(uint32_t level);
 
+/**
+ * Set the log level by string.
+ *
+ * @param level
```
Comment (KeithWiles) [inline lib/core/log/cne_log.h:70]: s/level/log_level/

File: lib/core/log/cne_log.c [hunk 8]
```diff
@@ -27,6 +28,22 @@ cne_log_set_level(uint32_t level)
         cne_loglevel = level;
 }
 
+int
+cne_log_set_level_str(char *log_level)
+{
+#define _(n, uc, lc)                                             \
```
Comment (KeithWiles) [inline lib/core/log/cne_log.c:34]: Check log_level not NULL.

File: lib/include/cne_strings.h [hunk 9]
```diff
@@ -260,6 +260,37 @@ cne_strqtok(char *str, const char *delim, char *argv[], int maxtokens)
...
+/**
+ * Convert characters in \p str to lowercase.
+ *
+ * @param str
+ *   String to convert to lowercase
+ */
+static __inline__ void
+cne_strtolower(char *str)
+{
+    for (int i = 0; i <= (int)(strlen(str)); i++)
```
Comment (KeithWiles) [inline lib/include/cne_strings.h:272]: Need to check str != NULL

File: lib/include/cne_strings.h [hunk 10]
```diff
@@ -260,6 +260,37 @@ cne_strqtok(char *str, const char *delim, char *argv[], int maxtokens)
...
+ * @param str
+ *   String to convert to uppercase
+ *
+ * @return
+ *   Upper case string
+ */
+static __inline__ char *
+cne_strtoupper(char *str)
+{
+    for (int i = 0; i <= (int)(strlen(str)); i++)
```
Comment (KeithWiles) [inline lib/include/cne_strings.h:288]: Check str != NULL

File: lib/core/log/cne_log.h [hunk 11]
```diff
@@ -36,9 +36,26 @@ enum {
     CNE_LOG_NOTICE,    /**< Normal but significant condition. */
     CNE_LOG_INFO,      /**< Informational.                    */
     CNE_LOG_DEBUG,     /**< Debug-level messages.             */
-    CNE_LAST_LOG
+    CNE_LOG_LAST
 };
 
+#define foreach_cndp_log_level \
```
Comment (maryamtahhan) [inline lib/core/log/cne_log.h:42]: we could consolidate into 1 set of log levels?

**Review by KeithWiles — CHANGES_REQUESTED** [review]
Klocwork will complain about NULL checks missing and a few minor nits.

File: lib/core/log/cne_log.h (hunk 11 above)
Comment (KeithWiles) [reply lib/core/log/cne_log.h:42]: ok

**Comment by maryamtahhan** [comment]
I was a little rusty on the c side :( but I think this should be in good place now :)

**Review by KeithWiles — APPROVED** [review]
LGTM

**Pull request merged**